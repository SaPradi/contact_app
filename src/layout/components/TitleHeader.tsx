
import '../styles/titleheader.css'

interface titleHeaderProps {
    title:string
}

const TitleHeader:React.FC<titleHeaderProps> = ({title}) => {
  return (
      <div className="header-section">
        <h1 className='header-section__title'>{title}</h1>
        <hr className='header-section__line'/>
      </div>
  )
}

export default TitleHeader
