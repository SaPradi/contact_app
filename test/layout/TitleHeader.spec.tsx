import React from "react";
import { render,screen} from "@testing-library/react";
import TitleHeader from '../../src/layout/components/TitleHeader';


describe('Testing in <TitleHeader>',()=>{

    const title = 'Contact'


    test('Should match with the snapshot', () => {
      
        const {container} = render( <TitleHeader title={title} />)

        expect( container ).toMatchSnapshot();

    })
    


    test('Should show a prop title', () => {


        render( <TitleHeader title={title} /> )

        const propsTitle = screen.getByRole('heading',{level:1});
        expect(propsTitle.textContent).toBe(title);

    })
    

})