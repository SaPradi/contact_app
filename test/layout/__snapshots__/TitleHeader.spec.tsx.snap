// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <TitleHeader> Should match with the snapshot 1`] = `
<div>
  <div
    class="header-section"
  >
    <h1
      class="header-section__title"
    >
      Contact
    </h1>
    <hr
      class="header-section__line"
    />
  </div>
</div>
`;
